rules/legacy_xcodeproj.bzl

```diff
@@ -315,8 +311,16 @@ def _xcodeproj_aspect_impl(target, ctx):
             transitive_targets.append(info)
 
         if test_host_target:
-            # Add the test_host_target to the targets to be added to the xcode project
-            test_host_direct_targets = test_host_target[_TargetInfo].direct_targets
+            test_host_direct_targets = []
+
+            for target in test_host_target[_TargetInfo].direct_targets:
+                # Add the test_host_target to the targets to be added to the xcode project
+                test_host_direct_targets.extend([target])
+
+                # Add the test_host_target's extensions to the targets to be added to the xcode project
+                for extension in target.extensions.to_list():
```

Comment:
Does this aggregate a transitive extension for a test bundle which depends on a hosted app?

Reply:
Yes. Previously, if I had `//tests/ios/unit-test:ExplicitHostedWithAppExtensions`, it wouldn't end up building in Xcode since the extensions for the test host would be missing.